test/utils/sagemaker_pipe.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python
+
+try:
+    # for python 3.x:
+    from urllib.parse import urlparse
+except ImportError:
+    # for python 2.x:
+    from urlparse import urlparse
+
+import argparse
+import boto3
+import errno
+import gzip
+import logging
+import os
+import shutil
+import stat
+
+NUM_EPOCHS = 5
+
+
+def run(args):
```

Comment:
Why do none of these methods have doc string?

Reply:
This was all copied from Ishaaq's [sagemaker pipe emulator](https://github.com/ishaaq/sagemaker-pipe/blob/master/sagemaker_pipe.py). I can try to add doc strings to these methods.

Reply:
Is there a reason this was copied? (Asking because I'm not as familiar with pipe mode).

If this does, please link original source in the file

Reply:
I copied this over in order to run it as a python script (which is how Ishaaq designed it to be used). I could also copy just the methods I need in order to create and delete the pipe, but I think this would be unnecessary work for little benefit.

Reply:
Ok. Please just refer to the original source in comments then!